core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -771,7 +772,14 @@ private[spark] class DAGScheduler(
     def visit(rdd: RDD[_]): Unit = {
       if (!visited(rdd)) {
         visited += rdd
-        val rddHasUncachedPartitions = getCacheLocs(rdd).contains(Nil)
+        val rddHasUncachedPartitions = try {
+          getCacheLocs(rdd).contains(Nil)
```

Comment:
Should we add a simple retry mechanism?

Reply:
treating it as not fully cached is a simple and reasonable fallback, and retry maybe an overkill here.